Editorial review of DCGM (#15198)
DataDog/integrations-core: dcgm/README.md

    ad.datadoghq.com/dcgm.init_configs: '[{}]'
    ad.datadoghq.com/dcgm.instances: |
      [
        {
          "openmetrics_endpoint": "http://%%host%%:9400/metrics"
        }
      ]
spec:
  containers:
    - name: dcgm
```

**Annotations v2** (for Datadog Agent v7.36+)

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: '<POD_NAME>'
  annotations:
    ad.datadoghq.com/dcgm.checks: |
      {
        "dcgm": {
          "init_config": {},
          "instances": [
            {
              "openmetrics_endpoint": "http://%%host%%:9400/metrics"
            }
          ]
        }
      }
spec:
  containers:
    - name: dcgm
```

<!-- xxz tabs xxx -->

2. [Restart the Agent][4].

### Validation

[Run the Agent's status subcommand][5] and look for `dcgm` under the Checks section.

### Adjusting Monitors

The monitors that come with this integration out of the box have some common-sense default values for their alert thresholds. For instance, the GPU temperature was determined based on [acceptable range for industrial devices][13].
However we recommend you check to make sure these values suit your particular needs.

## Data Collected

### Metrics

> See [metadata.csv][6] for a list of metrics provided by this integration.

Author: ```suggestion
See [metadata.csv][6] for a list of metrics that this integration provides.
```

### Events

The DCGM integration does not include any events.

### Service Checks

> See [service_checks.json][7] for a list of service checks provided by this integration.

Author: ```suggestion
See [service_checks.json][7] for a list of service checks that this integration provides.
```

## Troubleshooting

### Metric Mapping

> If you have added some metrics that don't appear in the [metadata.csv][6] above and appear in your account with the format `DCGM_FI_DEV_NEW_METRIC`, it is important to remap these metrics in the [dcgm.d/conf.yaml][3] configuration file:

Author: ```suggestion
If you have added some metrics that don't appear in the [metadata.csv][6] above but appear in your account with the format `DCGM_FI_DEV_NEW_METRIC`, it is important to remap these metrics in the [dcgm.d/conf.yaml][3] configuration file:
```

```yaml
    ## @param extra_metrics - (list of string or mapping) - optional
    ## This list defines metrics to collect from the `openmetrics_endpoint`, in addition to
    ## what the check collects by default. If the check already collects a metric, then
    ## metric definitions here take precedence. Metrics may be defined in 3 ways:
    ...
```

> The example below will append the part in `NEW_METRIC` to the namespace (`dcgm.`), giving `dcgm.new_metric`:

Author: ```suggestion
The example below appends the part in `NEW_METRIC` to the namespace (`dcgm.`), giving `dcgm.new_metric`:
```

```yaml
    extra_metrics:
    - DCGM_FI_DEV_NEW_METRIC: new_metric
```

> ### DCGM Field is Enabled but not Being Submitted?

Author: ```suggestion
### DCGM field is enabled but not being submitted?
```

> It may happen that you enable the collection of a field in `default-counters.csv` but it doesn't show up in Datadog, nor in fact if you make a `curl` request to `host:9400/metrics`.

Author: ```suggestion
It may happen that you've enabled the collection of a field in `default-counters.csv`, but it doesn't appear up in Datadog, even after making a `curl` request to `host:9400/metrics`.
```

> To figure out why this field is not being collected [dcgm-exporter devs recommend][14] looking at the file `var/log/nv-hostengine.log`.

Author: ```suggestion
To figure out why this field is not being collected, [dcgm-exporter devs recommend][14] looking at the file `var/log/nv-hostengine.log`.
```

Keep in mind that `dcgm-exporter` is a thin wrapper around lower-level libraries and drivers which do the actual reporting.

> ### Need Help?

Author: ```suggestion
### Need help?
```
